core/src/telemetry/tracing.ts

```diff
@@ -0,0 +1,363 @@
+ * attributes on the current OpenTelemetry span.
+ *
+ * @param params The parameters object containing invocationContext, eventId, and data.
+ */
+export function traceSendData({
+  invocationContext,
+  eventId,
+  data,
+}: TraceSendDataParams): void {
+  const span = trace.getActiveSpan();
+  if (!span) return;
+  
+  span.setAttributes({
+    'gcp.vertex.agent.invocation_id': invocationContext.invocationId,
+    'gcp.vertex.agent.event_id': eventId,
+  });
+
+  // Once instrumentation is added to the GenAI SDK, consider whether this
+  // information still needs to be recorded by the Agent Development Kit.
+
+  span.setAttribute('gcp.vertex.agent.data', safeJsonSerialize(data));
+}
+
+/**
+ * Builds a dictionary representation of the LLM request for tracing.
+ *
+ * This function prepares a dictionary representation of the LlmRequest
+ * object, suitable for inclusion in a trace. It excludes fields that cannot
+ * be serialized (e.g., function pointers) and avoids sending bytes data.
+ *
+ * @param llmRequest The LlmRequest object.
+ * @returns A dictionary representation of the LLM request.
+ */
+function buildLlmRequestForTrace(llmRequest: LlmRequest): Record<string, unknown> {
+  const result: Record<string, unknown> = {
+    model: llmRequest.model,
+    contents: [],
+  };
+
+  if (llmRequest.config) {
+    // Create a clean config object, pruning responseSchema to reduce noise size
+    const { responseSchema, ...cleanConfig } = llmRequest.config;
+    result.config = cleanConfig;
+  }
+
+  // We do not want to send bytes data to the trace.
+  result.contents = llmRequest.contents.map(content => ({
+    role: content.role,
+    parts: content.parts?.filter(part => !part.inlineData) || [],
+  }));
+
+  return result;
+}
+
+/**
+ * Binds an async generator to OpenTelemetry context for trace propagation.
+ * This is a temporary solution.	
+ * @param ctx - The OpenTelemetry context to bind the generator to
+ * @param generator - The async generator to be bound to the context
+ * 
+ * @returns A new async generator that executes all operations within the provided context
+ */
+function bindAsyncGenerator<T = unknown, TReturn = any, TNext = unknown>(
+  ctx: Context,
+  generator: AsyncGenerator<T, TReturn, TNext>,
+): AsyncGenerator<T, TReturn, TNext> {
+  return {
+    // Bind the next() method to execute within the provided context
+    next: context.bind(ctx, generator.next.bind(generator)),
+    
+    // Bind the return() method to execute within the provided context
+    return: context.bind(ctx, generator.return.bind(generator)),
+    
+    // Bind the throw() method to execute within the provided context
+    throw: context.bind(ctx, generator.throw.bind(generator)),
+
+    // Ensure the async iterator symbol also returns a context-bound generator
+    [Symbol.asyncIterator]() {
+      return bindAsyncGenerator(ctx, generator[Symbol.asyncIterator]());
+    },
+  };
+}
+
+export { tracer, bindAsyncGenerator };
```

Comment:
remove this and do `export function bindAsyncGenerator` and `export const tracer` instead, thanks